### PR TITLE
[MISC] Pass mutable to hibf ctor and avoid config copy

### PR DESCRIPTION
**Pull request by smehringer**
I think it is very surprising if something in the configuration changes although you have passed a `const &`. So I moved the checking and default setting to the very top and removed the copy from `layout::execute`.

I added a little bit documentation.

Supersedes #34

### CODE REVIEW

File: src/hierarchical_interleaved_bloom_filter.cpp [hunk 1]
```diff
@@ -198,8 +201,57 @@ void build_index(hierarchical_interleaved_bloom_filter & hibf,
...
+        else
+        {
+            cfg.tmax =
+                seqan::hibf::next_multiple_of_64(static_cast<uint16_t>(std::ceil(std::sqrt(cfg.number_of_user_bins))));
+        }
+    }
+    else if (cfg.tmax % 64 != 0)
+    {
+        cfg.tmax = seqan::hibf::next_multiple_of_64(cfg.tmax);
+        std::cerr << "[HIBF CONFIG WARNING]: Your requested number of technical bins was not a multiple of 64. "
```
Comment (smehringer) [inline src/hierarchical_interleaved_bloom_filter.cpp:246]: here I wasn't sure if we wanted to remove the warning? (in favor of documentation)

File: src/hierarchical_interleaved_bloom_filter.cpp [hunk 2]
```diff
@@ -198,8 +201,59 @@ void build_index(hierarchical_interleaved_bloom_filter & hibf,
...
+ *   `static_cast<uint16_t>(std::ceil(std::sqrt(cfg.number_of_user_bins)))`.
+ * * If seqan::hibf::config::tmax is **not** `0` but also not a multiple of 64, it is increased to the next multiple of
+ *   64 to avoid uneccessary space consumption (e.g. value `60` will be increased to `64` or `1000` increased to `1024`).
+ */
+void check_config_and_set_defaults(config & cfg)
 {
+    if (cfg.disable_estimate_union)
+        cfg.disable_rearrangement = true;
+
+    if (cfg.number_of_user_bins = 0)
```
Comment (eseiler) [inline src/hierarchical_interleaved_bloom_filter.cpp:226]:
```suggestion
    if (cfg.number_of_user_bins == 0)
```

Comment (smehringer) [reply src/hierarchical_interleaved_bloom_filter.cpp]: ups :D

File: src/hierarchical_interleaved_bloom_filter.cpp (hunk 1 above)
Comment (eseiler) [reply src/hierarchical_interleaved_bloom_filter.cpp:244]: I think it's OK to warn

**Review by eseiler — APPROVED** [review]

**Pull request merged**